src/installer/corehost/common/pal.unix.cpp

```diff
@@ -331,6 +331,38 @@ bool pal::get_temp_directory(pal::string_t& tmp_dir)
     return false;
 }
 
+bool pal::get_default_bundle_extraction_base_dir(pal::string_t& tmp_dir)
+{
+    if (!get_temp_directory(extraction_dir))
+    {
+        return false;
+    }
+
+    append_path(&extraction_dir, _X(".net"));
+    pal::string_t dotnetdir(extraction_dir);
+
+    append_path(&extraction_dir, getlogin());
+
+    if (is_read_write_able_directory(extraction_dir))
+    {
+        return true;
+    }
+
+    // Create $TMPDIR/.net accessible to everyone
+    if (::mkdir(dotnetdir, 0777) != 0 && errno != EEXIST)
+    {
+        return false;
+    }
+
+    // Create $TMPDIR/.net/username accessible only to the user
+    if (::mkdir(extraction_dir, 0700) != 0 && errno != EEXIST)
```

Comment:
I'd also set S_ISVTX (`01000` on Linux, but might be different on other systems) in these two `mkdir()` calls in addition to their access permissions.  This is the sticky bit.

With this bit set, only the owner of `extraction_dir` can rename or delete that directory.  (Otherwise, there's a potential race condition here: while you can only remove empty directories -- and need to recursively delete files from a directory -- a malicious process can remove the directory even before we were able to start extracting the files there.)

Reply:
Thanks @lpereira, I've fixed this on the `extraction_dir`, per our discussion.